### ISSUE TITLE
Fish freezes when tig segfaults at exit

### ISSUE THREAD

**Issue by verdie-g** [fish-shell/fish-shell#5596]
Hi,

It seems like [tig](https://github.com/jonas/tig) version 2.4.1 segfaults on exit and it makes fish freeze. No problem with bash.

Steps to reproduce:
```
git clone https://github.com/jonas/tig
cd tig
git checkout tig-2.4.1
make configure
./configure
make -j 4
echo | src/tig
```

**Comment by ridiculousfish** [comment]
I can reproduce. Note it also segfaults with bash but bash recovers.

**Comment by ridiculousfish** [comment]
Bisects to af0c8d51e0022e747302128abcb77e8beaef4d42

**Comment by mqudsi** [comment]
It also used to happen when piping `git ...` to `tig`, which is a far more common thing to do.

**Comment by mqudsi** [comment]
Fix cherry-picked for 3.0.1 as 4dfaa33d95a2cc16f4a8eb757948bd9053dfbb71
